packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -729,6 +729,10 @@ function convertN8nRequestToAxios(n8nRequest: IHttpRequestOptions): AxiosRequest
 		axiosRequest.headers['User-Agent'] = 'n8n';
 	}
 
+	if (n8nRequest.validateStatus !== undefined) {
+		axiosRequest.validateStatus = n8nRequest.validateStatus;
+	}
```

Comment:
> By default all of the request helpers treat non-success codes as errors. To get back the error response with headers I needed to add the Axios `validateStatus` option to the `IHttpRequestOptions`. In developing nodes I've sometimes wondered if I could get all responses and decide what constitutes a failure condition. Is this something I've missed that is already possible?

@krynble If you can confirm that you meant for this to be done with `validateStatus`.

Reply:
Hey @ivov as you mentioned, the new `httpRequest` helper does not provide a way to return the response data when a request is not 2xx.

I believe the best way is to provide a new option named `ignoreHttpStatusErrors` or another name with a behavior similar to the existing `returnFullResponse` option that can be see in the `httpRequest` function in this same file.

Reply:
@ivov, @krynble here's a version using `ignoreHttpStatusErrors` instead:
https://github.com/n8n-io/n8n/compare/master...pemontto:wise-sca2?expand=1

Main change is 
```js
if (n8nRequest.ignoreHttpStatusErrors) {
    axiosRequest.validateStatus = () => true;
}
```